hack/build/build_image.sh

```diff
@@ -26,7 +26,14 @@ else
   CONTAINER_CMD=docker
 fi
 
-${CONTAINER_CMD} build . -f ./Dockerfile -t "${out_image}" \
+if [ -n "$OPERATOR_DEV_MAC" ];then
+  echo "overriding platform to ${OPERATOR_DEV_MAC}"
+  PLATFORM="--platform=${OPERATOR_DEV_MAC} "
+else
+  PLATFORM=""
+fi
+
+${CONTAINER_CMD} build ${PLATFORM}. -f ./Dockerfile -t "${out_image}" \
```

Comment:
```suggestion
${CONTAINER_CMD} build "${PLATFORM}" -f ./Dockerfile -t "${out_image}" \
```

Reply:
can't omit the `.` in podman build cmd, right?

Reply:
aha you have typo I guess.

Reply:
so to be more percise:
```
podman build --platform=linux/amd64. -f ./Dockerfile --tag t
Error: invalid platform syntax for --platform="linux/amd64.": "amd64." is an invalid component of "linux/amd64.": platform specifier component must match "^[A-Za-z0-9_-]+$": invalid argument
```

Reply:
so better to add space to bash, instead of add to PLATFORM variable, again to not confuse others.

Reply:
```diff
diff --git hack/build/build_image.sh hack/build/build_image.sh
index ae0637d4..917ba527 100755
--- hack/build/build_image.sh
+++ hack/build/build_image.sh
@@ -28,12 +28,12 @@ fi

 if [ -n "$OPERATOR_DEV_MAC" ];then
   echo "overriding platform to ${OPERATOR_DEV_MAC}"
-  PLATFORM="--platform=${OPERATOR_DEV_MAC} "
+  PLATFORM="--platform=${OPERATOR_DEV_MAC}"
 else
   PLATFORM=""
 fi

-${CONTAINER_CMD} build ${PLATFORM}. -f ./Dockerfile -t "${out_image}" \
+${CONTAINER_CMD} build ${PLATFORM} . -f ./Dockerfile -t "${out_image}" \
   --build-arg "GO_LINKER_ARGS=${go_linker_args}" \
   --build-arg "GO_BUILD_TAGS=${go_build_tags}" \
   --label "quay.expires-after=14d"
```

Reply:
your idea is okay, but space is better to add to bash directly, not to a variable. If we change our minds in the future, it's easy to replace instead of playing with spaces.

Reply:
> it actually mimics `DOCKER_DEFAULT_PLATFORM` env var...